geomstats/_backend/numpy/__init__.py

```diff
@@ -118,6 +118,24 @@ def get_mask_i_float(i, n):
     return mask_i_float
 
 
+def _is_boolean(x):
+    if isinstance(x, bool):
+        return True
+    if isinstance(x, (tuple, list)):
+        return _is_boolean(x[0])
```

Comment:
Why looking at only the first element of the tuple/list? It is assumed that all elements have the same type?

Reply:
Just saw your answer in the former PR:

I made the assumption that either all elements are booleans or no one is. I don't think it makes sense to make the code heavier and longer to run to be able to account for cases where booleans would be mixed with non-boolean values. I am not even sure Numpy handles them.

Reply:
I made the assumption that either all elements are booleans or no one is. I don't think it makes sense to make the code heavier and longer to run to be able to account for cases where booleans would be mixed with non-boolean values. I am not even sure Numpy handles them.